test/e2e/README.md

```diff
@@ -1,11 +1,124 @@
 # End to End testing
 
+# ToC
+- [How it works](#how-it-works)
+  * [Setup instance](#setup-instance)
+  * [Run instance](#run-instance)
+  * [Command line](#command-line)
+    * [AWS development](#aws-development)
+    * [Locally](#locally)
+- [Argo workflow](#argo-workflow)
+  * [argo assertion](#argo-assertion)
+  * [argo container](#argo-container)
+- [Upgrade](#upgrade---bump)
+  * [Hyperkube](#bump-hyperkube-version)
+  * [Pupernetes](#bump-pupernetes---bump-the-version-of-p8s)
+  * [Argo](#bump-argo)
+  * [Flatcar - Container Linux](#bump-coreos-container-linux---kinvolk-flatcar)
+
+# How it works
+
+There are 3 main directories:
+- [argo-workflows](./argo-workflows)
+    Specification of the end to end testing
+
+- [containers](./containers) 
+    Custom container images needed within the workflows
+
+- [scripts](./scripts)
+    * setup-instance
+        Entrypoint and scripts dedicated for environments (locally, AWS dev, AWS gitlab)
+    * run-instance
+        Scripts executed in the argo-machine (locally, AWS instance)
+
+## setup-instance
+
+<img src="docs/setup-instance.svg" width="350">
+
+## run-instance
+
+<img src="docs/run-instance.svg" width="100">
+
+## Command line
+
+### AWS development
+
+```bash
+$ cd ${GOPATH}/src/github.com/DataDog/datadog-agent 
+$ aws-vault exec ${DEV} -- inv -e e2e-tests -t dev --image datadog/agent-dev:master
+$ echo $?
+```
+
+### Locally
+
+```bash
+$ sudo pupernetes daemon run /opt/sandbox --job-type systemd 
+$ cd ${GOPATH}/src/github.com/DataDog/datadog-agent 
+$ inv -e e2e-tests -t local --image datadog/agent-dev:master
+$ echo $?
+```
+
+# Argo workflow
+
+The argo documentation is available [here](https://applatix.com/open-source/argo/docs/examples.html), there are a lot of example [here](https://github.com/argoproj/argo/tree/master/examples) too.
+
+## Argo assertion
+
+To assert something in an argo workflow, you need to create a mongodb query:
+```yaml
+name: find-kubernetes-state-deployments
+activeDeadlineSeconds: 200
+script:
+  image: mongo:3.6.3
+  command: [mongo, "fake-datadog.default.svc.cluster.local/datadog"]
+  source: |
+    while (1) {
+      var nb = db.series.find({
+      metric: "kubernetes_state.deployment.replicas_available", 
+      tags: {$all: ["namespace:default", "deployment:fake-datadog"] }, 
+      "points.0.1": { $eq: 1} });      
+      print("find: " + nb)
+      if (nb != 0) {
+        break;
+      }
+      prevNb = nb;
+      sleep(30000);
```

Comment:
This seems like a large value between retries: `sleep(30000)`.

A lot of our steps are using `sleep(2000)`.

What do you think?

Reply:
As we discussed, a large value is needed to observe a removed check.
A short one is better on query, to speed the tests up.